pkg/apis/vsphere/v1/template_genclient.go

```diff
@@ -0,0 +1,97 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+var (
+	// ErrTemplateOperationNotSupported is returned if operations other than Get and List are performed
+	ErrTemplateOperationNotSupported = fmt.Errorf("%w: Template only support Get and List operations", api.ErrOperationNotSupported)
+)
+
+// EndpointURL returns the URL where to retrieve objects of type Template (only Get and List operations supported atm).
+func (t *Template) EndpointURL(ctx context.Context) (*url.URL, error) {
+	_, err := t.operationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return url.ParseRequestURI(
+		fmt.Sprintf("/api/vsphere/v1/provisioning/templates.json/%s/%s?page=1&limit=1000", t.Location.Identifier, t.Type),
+	)
+}
+
+// HasPagination disables pagination for Template API (not supported by engine)
+func (t *Template) HasPagination(ctx context.Context) (bool, error) {
+	return false, nil
+}
+
+// FilterRequestURL removes the Identifier from URL on Get operations (template needs to be parsed from list response)
+func (t *Template) FilterRequestURL(ctx context.Context, url *url.URL) (*url.URL, error) {
+	op, err := t.operationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if op == types.OperationGet {
+		url.Path = path.Dir(url.Path)
+	}
+
+	return url, nil
+}
+
+// DecodeAPIResponse is used to filter a single template on Get operations
+func (t *Template) DecodeAPIResponse(ctx context.Context, data io.Reader) error {
+	op, err := t.operationFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	if op == types.OperationGet {
+		tpl, err := t.extractTemplateByID(data)
+		if err != nil {
+			return err
+		}
+		*t = *tpl
+		return nil
+	}
+
+	return json.NewDecoder(data).Decode(t)
+}
+
+func (t *Template) extractTemplateByID(data io.Reader) (*Template, error) {
+	var templates []*Template
+	err := json.NewDecoder(data).Decode(&templates)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, template := range templates {
+		if template.Identifier == t.Identifier {
+			return template, nil
+		}
+	}
+
+	return nil, ErrTemplateNotFound
```

Comment:
`ErrTemplateNotFound` is for "named template not found" but this case here isn't about named templates. Probably just return a `pkg/api.ErrNotFound` here, maybe wrap with "Template with given Identifier not found in response".